7194 : No local JVMs found message could be more helpful

docwarems opened the pull request:
The message refers to the FAQ to question 
"Q: I cannot see any locally running JVMs in the JVM browser!" 
Suggestion for new answer
A3: Are you running a JDK <11? You must specify a JDK >=11 either via PATH of by adding "-vm" parameter in jmc.ini (before -vmargs parameter; value is path to JDK bin folder)

<!-- Anything below this marker will be automatically updated, please do not edit manually! -->
---------
### Progress
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JMC-7194](https://bugs.openjdk.java.net/browse/JMC-7194): No local JVMs found message could be more helpful


### Reviewers
 * [Marcus Hirt](https://openjdk.java.net/census#hirt) (@thegreystone - **Reviewer**)


### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jmc pull/232/head:pull/232` \
`$ git checkout pull/232`

Update a local copy of the PR: \
`$ git checkout pull/232` \
`$ git pull https://git.openjdk.java.net/jmc pull/232/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 232`

View PR using the GUI difftool: \
`$ git pr show -t 232`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jmc/pull/232.diff">https://git.openjdk.java.net/jmc/pull/232.diff</a>

</details>

docwarems commented:
/signed

thegreystone commented:
Hi there!

There is a suggestion to update the FAQ (https://github.com/openjdk/jmc/blob/master/application/org.openjdk.jmc.docs/html/GUID-3EC7E6C3-6A64-4659-B99E-CA675A9F6E0A.htm), there is however no update to that html file present in the PR.

docwarems commented:
> There is a suggestion to update the FAQ
OK, but the content of ths repo file is different from the FAQs content I found by Google, to which my PR description refers.

https://wiki.openjdk.java.net/display/jmc/JMC+FAQ
https://community.oracle.com/tech/developers/discussion/2579717/jmc-frequently-asked-questions

thegreystone commented:
Feel free to update the JMC-included documentation if you feel it improves it. :) The project itself mostly concerns itself with the included documentation, not the Oracle-hosted FAQ.

docwarems commented:
OK, IMO we have this situation:

https://wiki.openjdk.java.net/display/jmc/JMC+FAQ (last updated 02/2020) is the location for publishing the FAQ currently and in the future. It's a Confluence system, and there's no way to sync this with a Git document, because Confluence has it's own versioning.
That means the purpose of the repo FAQ (last update 01/2019) could be not more than for saving the raw content (the HTML format is only used for better reading, otherwise it could be even plain text).

Is my view correct?

To me the published FAQ looks much more useful than the repo FAQ. If you ask me, I would replace the repo with the content from the published FAQ, and add the answer from this PR.

At least we have the content of both in sync then.

thegreystone commented:
Does this render OK in the help browser?

1. Start JMC.
2. Go to Help | JDK Mission Control Help.
3. Click Frequently Asked Questions (at the bottom).

docwarems commented:
> Does this render OK in the help browser?
Probably not ;-)

But from my previous comments it could have become obviuos that I wasn't aware of the repo FAQ shown elsewhere. Now that I know that it's parts of the included help, I reverted the commit and added just my new answer.

How will we get rid of the extra commits? Have I to re-push the branch again or is there a squash functionality when the PR will be merged. Sorry, I need help here because it's my first contribution to a GitHub project.

Gunde commented:
> How will we get rid of the extra commits? Have I to re-push the branch again or is there a squash functionality when the PR will be merged. Sorry, I need help here because it's my first contribution to a GitHub project.

 When the PR is integrated all commits will be squashed, so you don't really need to do anything particular here.

thegreystone commented:
As Henrik said - it's all part of the Skara workflow. It will all automatically be squashed into one neat commit in the end. :)

thegreystone commented on application/org.openjdk.jmc.browser/src/main/resources/org/openjdk/jmc/browser/views/messages.properties:
```diff
@@ -57,7 +57,7 @@ JVMBrowserView_NO_LOCAL_JVMS_TITLE=No local JVMs detected
 # JVMBrowserView_NO_LOCAL_JVMS_WARN_CAUSE has newlines placed so that the rows doesn't get too long. It would be
 # great if the translation also contains newlines so the rows get about the same length.
-JVMBrowserView_NO_LOCAL_JVMS_WARN_CAUSE=Local JVMs will not be detected if the JDK Mission Control Client is run with a JRE instead of a JDK, or if the hsperfdata_<user> folder in TMP is not writable. See the FAQ help section for more information.
+JVMBrowserView_NO_LOCAL_JVMS_WARN_CAUSE=Local JVMs will not be detected if the JDK Mission Control Client is run with a JRE instead of a JDK, if the JDK is <11, or if the hsperfdata_<user> folder in TMP is not writable. See the FAQ help section for more information.
```
Perhaps instead spell out the less than and expand to JDK version? Like so ", if the JDK version is less than 11, "

docwarems replied:
Ok, this one too. Saw this first comment to late -> two commits

thegreystone replied:
No problem. You can have a gazillion commits - the friendly neighborhood Skara-bot will automatically squash. :)

thegreystone commented on application/org.openjdk.jmc.docs/html/GUID-3EC7E6C3-6A64-4659-B99E-CA675A9F6E0A.htm:
```diff
@@ -79,6 +79,8 @@ <h1 id="JMCOH-GUID-3EC7E6C3-6A64-4659-B99E-CA675A9F6E0A" class="sect1"><span cla
 </li>
 <li><p>Make sure that there is a directory named <code>hsperfdata_<span class="codeinlineitalic">username</span></code> in the system's <code>tmp</code> directory, that it is writable by the user running JMC, and that the file system supports access control lists (ACLs).</p>
 </li>
+<li><p>Make sure you are you running a JDK >= 11. You can set the JDK bin directory either by&#160;<tt>PATH</tt>&#160;environment variable, or by adding&#160;<tt>-vm</tt>&#160;parameter to&#160;<tt>jmc.ini</tt>&#160;file (must be inserted before&#160;<tt>-vmargs</tt>&#160;section).</p>
```
How about making explicit that it is Mission Control that should be running on the proposed version? "Make sure you are running Mission Control on "

docwarems replied:
+1 done

thegreystone commented:
Hi @docwarems! Please read the comments above from the bot. It contains instruction on how to proceed with integrating the PR. Let me know if you need any help!

docwarems commented:
I have problems fixing the user name for my commit as described by the bot.
The command gives me this error with my Git client: 
fatal: Option -m cannot be combined with -c/-C/-F.

If I omit the -m option it gives me the error
fatal: could not lookup commit user.name=Michael S<C3><BC>dkamp

This is my user name as obtained from "git config --list". As you can see it has an escaped german character. I also tried without success with the unescaped name and with umlaut character replaced by ASCII.
What can I do?

thegreystone commented:
Don’t worry too much then. If you are happy with how your name looks in the commits, then simply write /integrate in a comment.

docwarems commented:
/integrate

thegreystone commented:
/sponsor
